src/sql/workbench/services/accountManagement/browser/accountManagementService.ts

```diff
@@ -204,9 +207,11 @@ export class AccountManagementService implements IAccountManagementService {
 
 		return this.doWithProvider(account.key.providerId, async (provider) => {
 			let refreshedAccount = await provider.provider.refresh(account);
-			if (self.isCanceledResult(refreshedAccount)) {
-				// Pattern here is to throw if this fails. Handled upstream.
-				throw new Error(localize('refreshFailed', "Refresh account was canceled by the user"));
+			if (self.isPromptFailedResult(refreshedAccount)) {
+				if (refreshedAccount.canceled) {
```

Comment:
If there's an error you should be handling that - otherwise you'll have a problem when you get to line 219 and try to add the (undefined) account